### PR TITLE
Lesson 6 vending machine assignment

### PR DESCRIPTION
**Pull request by Boinciel**

### CODE REVIEW

File: src/assignments/PAUL_OMAKI/assignment06.clj [hunk 1]
```diff
@@ -0,0 +1,155 @@
+(ns assignments.PAUL-OMAKI.assignment06)
+
+
+
+(def vending-machine "Vending machine base state.
+                      
+                      `:state :start`
+                      `:coins []`"
+  {:state :start :coins []})
```
Comment (privisi) [inline src/assignments/PAUL_OMAKI/assignment06.clj:11]: normally the comment is on a line below the def

File: src/assignments/PAUL_OMAKI/assignment06.clj [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  "Checks the state of the vending machine to see if there is enough to purchase the soda.
+   If there is, dispenses soda and refunds whatever is left in the machine."
+  ([machine]
+   (dispense-if-enough-money machine :cola) )
+  ([machine drink]
+   (if (< (coin-value machine) (price-of-soda drink))
+     (do 
+       (println "Not enough money.")
+       machine)
+     (if (< 0 (- (coin-value machine) (price-of-soda drink)))
```
Comment (privisi) [inline src/assignments/PAUL_OMAKI/assignment06.clj]: neg?

File: src/assignments/PAUL_OMAKI/assignment06.clj [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+       (do                               ; We have enough money! Too much of it, in fact!
+         (println "Disbursing delicious drink: " drink)
+         (println "Refunding " (- (coin-value machine) (price-of-soda drink)) " cents.")
```
Comment (privisi) [inline src/assignments/PAUL_OMAKI/assignment06.clj]: repeated calculation, you can put this in a let

File: src/assignments/PAUL_OMAKI/assignment06.clj [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  (println "Ready state, but " event " activated.")
+  (println "Refunding " (:coins machine))
+  vending-machine)
+
+(defmethod handle [:ready :coin] [machine event]
+  (println "Getting a coin in the ready state: " (:coin-value event) " cents added.")
+           (let [machine (-> machine
+                             (assoc :state :ready)
+                             (update :coins conj (:coin-value event)))]
+             machine))
```
Comment (privisi) [inline src/assignments/PAUL_OMAKI/assignment06.clj:107]: weird indenting

File: src/assignments/PAUL_OMAKI/assignment06.clj [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+(defmethod handle [:ready :button] [machine event]
+  (println "Button pressed: " (:button-value event))
+  (let [machine  machine
```
Comment (privisi) [inline src/assignments/PAUL_OMAKI/assignment06.clj]: why?

File: src/assignments/PAUL_OMAKI/assignment06.clj [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+        drink    (:button-value event)]
```
Comment (privisi) [inline src/assignments/PAUL_OMAKI/assignment06.clj:96]: just destructure?

**Review by privisi — CHANGES_REQUESTED** [review]
Changes requested

File: src/assignments/PAUL_OMAKI/assignment06.clj (hunk 1 above)
Comment (Boinciel) [reply src/assignments/PAUL_OMAKI/assignment06.clj:11]: It should be in the correct place now.

File: src/assignments/PAUL_OMAKI/assignment06.clj (hunk 2 above)
Comment (Boinciel) [reply src/assignments/PAUL_OMAKI/assignment06.clj]: It only triggers if the surplus is positive.

File: src/assignments/PAUL_OMAKI/assignment06.clj (hunk 4 above)
Comment (Boinciel) [reply src/assignments/PAUL_OMAKI/assignment06.clj:107]: Fixed, thanks.

File: src/assignments/PAUL_OMAKI/assignment06.clj (hunk 5 above)
Comment (Boinciel) [reply src/assignments/PAUL_OMAKI/assignment06.clj]: Good point, removed.

File: src/assignments/PAUL_OMAKI/assignment06.clj (hunk 2 above)
Comment (privisi) [reply src/assignments/PAUL_OMAKI/assignment06.clj]: there is a function called neg? which does what you want

File: src/assignments/PAUL_OMAKI/assignment06.clj (hunk 3 above)
Comment (Boinciel) [reply src/assignments/PAUL_OMAKI/assignment06.clj]: Adjusted code for readability.

File: src/assignments/PAUL_OMAKI/assignment06.clj (hunk 2 above)
Comment (Boinciel) [reply src/assignments/PAUL_OMAKI/assignment06.clj]: Ah, I see. Thanks for the heads up. Fixed.

**Review by privisi — APPROVED** [review]

**Pull request merged**